In [1]:
pip install folium

Note: you may need to restart the kernel to use updated packages.


In [2]:
import numpy as np
import pandas as pd 
import seaborn as sns 
import matplotlib.pyplot as plt 
import warnings 
warnings.filterwarnings('ignore')

In [3]:


# Load the dataset with the correct encoding
df = pd.read_csv('Global_terrorist.csv', encoding='ISO-8859-1')



In [4]:
df

eventid  iyear  imonth  iday approxdate  extended resolution  \
0       197000000001   1970       7     2        NaN         0        NaN   
1       197000000002   1970       0     0        NaN         0        NaN   
2       197001000001   1970       1     0        NaN         0        NaN   
3       197001000002   1970       1     0        NaN         0        NaN   
4       197001000003   1970       1     0        NaN         0        NaN   
...              ...    ...     ...   ...        ...       ...        ...   
181686  201712310022   2017      12    31        NaN         0        NaN   
181687  201712310029   2017      12    31        NaN         0        NaN   
181688  201712310030   2017      12    31        NaN         0        NaN   
181689  201712310031   2017      12    31        NaN         0        NaN   
181690  201712310032   2017      12    31        NaN         0        NaN   

        country         country_txt  region  ... addnotes  \
0            58  Dominican Republic       2  ...      NaN   
1           130              Mexico       1  ...      NaN   
2           160         Philippines       5  ...      NaN   
3            78              Greece       8  ...      NaN   
4           101               Japan       4  ...      NaN   
...         ...                 ...     ...  ...      ...   
181686      182             Somalia      11  ...      NaN   
181687      200               Syria      10  ...      NaN   
181688      160         Philippines       5  ...      NaN   
181689       92               India       6  ...      NaN   
181690      160         Philippines       5  ...      NaN   

                                                   scite1  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Somalia: Al-Shabaab Militants Attack Army Che...   
181687  "Putin's 'victory' in Syria has turned into a ...   
181688  "Maguindanao clashes trap tribe members," Phil...   
181689  "Trader escapes grenade attack in Imphal," Bus...   
181690  "Security tightened in Cotabato following IED ...   

                                                   scite2  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Highlights: Somalia Daily Media Highlights 2 ...   
181687  "Two Russian soldiers killed at Hmeymim base i...   
181688                                                NaN   
181689                                                NaN   
181690  "Security tightened in Cotabato City," Manila ...   

                                                   scite3  \
0                                                     NaN   
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
181686  "Highlights: Somalia Daily Media Highlights 1 ...   
181687  "Two Russian servicemen killed in Syria mortar...   
181688                                                NaN   
181689                                                NaN   
181690                                                NaN   

                        dbsource  INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0                           PGIS        0         0        0       0      Na

In [5]:
filter1 = df[df['region_txt'] == "Middle East & North Africa"]

In [6]:
filter2 = filter1[filter1['iyear'] > 2000]

In [7]:
filter2['attacktype1_txt'].value_counts()

attacktype1_txt
Bombing/Explosion                      26819
Armed Assault                           6980
Hostage Taking (Kidnapping)             2316
Assassination                           2157
Unknown                                 1597
Facility/Infrastructure Attack           698
Unarmed Assault                          102
Hostage Taking (Barricade Incident)       70
Hijacking                                 67
Name: count, dtype: int64

In [8]:
filter3 = filter2[filter2['attacktype1_txt'] == 'Bombing/Explosion']

In [9]:
final_filter = filter3.groupby(['latitude', 'longitude']).size().reset_index(name = 'incident_count')

<h1>Use of Folium For GEOSPATIAL</h1>

In [10]:
import folium
from folium.plugins import HeatMap

<h2>Converting Pandas Data Frame into Python List</h2>

In [11]:
heatmap_data = final_filter[['latitude', 'longitude', 'incident_count']].values.tolist()

<h2>Creating a Base map</h2>

In [12]:
my_map = folium.Map(location = [30,40], zoom_start = 4)

<h2>Add Heatmap Layer</h2>

In [13]:
HeatMap(heatmap_data).add_to(my_map)

In [14]:
my_map.save('terrorism_bombing.html')
print("data save to terrorism_bombing.html")

data save to terrorism_bombing.html


In [15]:
my_map

<h1>Nationality Analysis on Philippines</h1>

In [16]:
Philippines = df[df['country_txt'] == "Philippines"]

In [17]:
Philippines['attacktype1_txt'].value_counts()

attacktype1_txt
Bombing/Explosion                      2213
Armed Assault                          2158
Assassination                          1099
Hostage Taking (Kidnapping)             602
Facility/Infrastructure Attack          452
Unknown                                 293
Hostage Taking (Barricade Incident)      53
Hijacking                                34
Unarmed Assault                           4
Name: count, dtype: int64

In [18]:
Philippines2 = Philippines[Philippines['iyear'] > 2015]
Philippines3 = Philippines[Philippines['attacktype1_txt'] == 'Armed Assault']

In [19]:
final_Philippines = Philippines3.groupby(['latitude', 'longitude']).size().reset_index(name = 'incident_count')

In [20]:
heatmap_data = final_Philippines[['latitude', 'longitude', 'incident_count']].values.tolist()

In [21]:
my_mapPhilippines = folium.Map(location = [30,40], zoom_start = 4)

In [22]:
HeatMap(heatmap_data).add_to(my_mapPhilippines)

In [23]:
my_mapPhilippines.save('Armed Assault.html')
print("data save to Armed Assault.html")

data save to Armed Assault.html


In [24]:
my_mapPhilippines

<h1>Insight no.1</h1>
<h2>From 2015 to 2017, there were 1,324 Armed Assault incidents in the Philippines. This shows a significant number of attacks over the years.The heatmap above shows where these occurred and provides a visual representation of the locations most affected.</h2>

<h1>Attack Type analysis on Iraq</h1>

In [25]:
Iraq = df[df['country_txt'] == "Iraq"]

In [26]:
Iraq['attacktype1_txt'].value_counts()

attacktype1_txt
Bombing/Explosion                      18286
Armed Assault                           3155
Assassination                           1287
Hostage Taking (Kidnapping)              946
Unknown                                  765
Facility/Infrastructure Attack           154
Hostage Taking (Barricade Incident)       24
Hijacking                                 13
Unarmed Assault                            6
Name: count, dtype: int64

In [27]:
Iraq2 = Iraq[Iraq['attacktype1_txt'] == 'Assassination']

In [28]:
final_Iraq = Iraq2.groupby(['latitude', 'longitude']).size().reset_index(name = 'incident_count')

In [29]:
heatmap_data = final_Iraq[['latitude', 'longitude', 'incident_count']].values.tolist()

In [30]:
my_mapIraq = folium.Map(location = [30,40], zoom_start = 4)

In [31]:
HeatMap(heatmap_data).add_to(my_mapIraq)

In [32]:
my_mapIraq.save('IraqAssassination.html')
print("data save to IraqAssassination.html")

data save to IraqAssassination.html


In [33]:
my_mapIraq

<h1>Insight no.2</h1>
<h2>From 1970 to 2017, there were 1,287 Bombing/Explosion incidents in the Iraq. This shows a significant number of attacks over the years.The heatmap above shows where these occurred and provides a visual representation of the locations most affected.</h2>

In [41]:
US = df[df['country_txt'] == "United States"]
US2 = US[US['attacktype1_txt'] == 'Bombing/Explosion']
US2 = US2[US2['nkill'] >= 10]
US3 = US2.groupby(['latitude', 'longitude']).size().reset_index(name='incident_count')
heatmap_data = US3[['latitude', 'longitude', 'incident_count']].values.tolist()
my_mapUS = folium.Map(location=[30, 40], zoom_start=4)
HeatMap(heatmap_data).add_to(my_mapUS)
my_mapUS.save('USterroristAttack.html')
print("data saved to USterroristAttack.html")
my_mapUS

data saved to USterroristAttack.html


<h1>Insight no.3</h1>
<h2>From 1970 to 2017 in the United States, the most common type of attack was Bombing/Explosion, with 1,383 incidents, indicating a frequent use of explosives as a method of violence during this period. Facility/Infrastructure Attacks followed closely with 864 cases, showing a significant focus on disrupting key structures or services. The heatmap of where it happened helps visualize the geographic distribution of these events, highlighting regional hotspots and areas most affected by such violence.</h2>

<h1>Regional Analysis of South Asia<h1>

In [ ]:
South_Asia_df = df[df['region_txt'] == "South Asia"]

In [ ]:
South_Asia_df['nkill'].value_counts()

In [ ]:
South_Asia_df2 = South_Asia_df[South_Asia_df['attacktype1_txt'] == 'Assassination']

In [ ]:
final_South_Asia = South_Asia_df2.groupby(['latitude', 'longitude']).size().reset_index(name = 'incident_count')

In [ ]:
heatmap_data = final_South_Asia[['latitude', 'longitude', 'incident_count']].values.tolist()

In [ ]:
my_mapSouth_Asia = folium.Map(location = [30,40], zoom_start = 4)

In [ ]:
HeatMap(heatmap_data).add_to(my_mapSouth_Asia)

In [ ]:
my_mapSouth_Asia.save('SouthAsiaterroristAttack.html')
print("data save to SouthAsiaterroristAttack.html")

In [ ]:
my_mapSouth_Asia

<h1>Insight no.4</h1>
<h2>Between 1970 and 2017, South Asia experienced 4,301 assassination incidents across various countries. This reflects a substantial number of attacks over the years. The heatmap above visually highlights the regions most impacted by these incidents.</h2>

<h1>Village/City/Town/Suburb target Analysis on Sub-Sarahan Africa </h1>

In [ ]:
SS_Africa_df = df[df['region_txt'] == "Sub-Saharan Africa"]

In [ ]:
SS_Africa_df['targsubtype1_txt'].value_counts()

In [ ]:
SS_Africa_df2 = SS_Africa_df[SS_Africa_df['targsubtype1_txt'] == 'Village/City/Town/Suburb']

In [ ]:
final_SS_Africa_df = SS_Africa_df2.groupby(['latitude', 'longitude']).size().reset_index(name = 'incident_count')

In [ ]:
heatmap_data = final_SS_Africa_df[['latitude', 'longitude', 'incident_count']].values.tolist()

In [ ]:
my_mapSS_Africa = folium.Map(location = [30,40], zoom_start = 4)

In [ ]:
HeatMap(heatmap_data).add_to(my_mapSS_Africa)

In [ ]:
my_mapSS_Africa.save('SS_AfricaterroristAttack.html')
print("data save to SS_AfricaterroristAttack.html")

In [ ]:
my_mapSS_Africa

<h1>Insight no.5</h1>
<h2>Based on the data provided, there were 1,846 incidents that Village/City/Town/Suburb is targeted in Sub-Saharan Africa region between 1970 and 2017. This indicates that the region experienced a consistent threat of such attacks over nearly five decades. The heatmap above shows where these incidents occurred, highlighting areas most affected by this type of violence.</h2>